docs/swift-3-api.diff

```diff
@@ -0,0 +1,399 @@
++    public static let fileFormatUpgradeRequired: Code = .fileFormatUpgradeRequired
++    public static let addressSpaceExhausted: Code = .addressSpaceExhausted
++    public static let schemaMismatch: Code = .schemaMismatch
++    public var code: Code
++    public var _nsError: NSError
++    public init(_nsError error: NSError)
++public typealias _ErrorType = RLMError
++    public static var _nsErrorDomain = RLMErrorDomain
+ public class LinkingObjectsBase: NSObject, NSFastEnumeration
+-    public func countByEnumeratingWithState(state: UnsafeMutablePointer<NSFastEnumerationState>, objects buffer: AutoreleasingUnsafeMutablePointer<AnyObject?>, count len: Int) -> Int
++    public func countByEnumerating(with state: UnsafeMutablePointer<NSFastEnumerationState>, objects buffer: AutoreleasingUnsafeMutablePointer<AnyObject?>!, count len: Int) -> Int
+ public final class LinkingObjects<T: Object>: LinkingObjectsBase
+     public subscript(index: Int) -> T
+     public typealias Element = T
+     public var realm: Realm? 
+-    public var invalidated: Bool 
++    public var isInvalidated: Bool
+     public var count: Int 
+     public init(fromType type: T.Type, property propertyName: String)
+     public override var description: String
+-    public func indexOf(object: T) -> Int?
+-    public func indexOf(predicate: NSPredicate) -> Int?
+-    public func indexOf(predicateFormat: String, _ args: AnyObject...) -> Int?
++    public func index(of object: T) -> Int?
++    public func indexOfObject(for predicate: NSPredicate) -> Int?
++    public func indexOfObject(for predicateFormat: String, _ args: Any...) -> Int?
+     public var first: T? 
+     public var last: T? 
+-    public override func valueForKey(key: String) -> AnyObject?
+-    public override func valueForKeyPath(keyPath: String) -> AnyObject?
+-    public override func setValue(value: AnyObject?, forKey key: String)
+-    public func filter(predicateFormat: String, _ args: AnyObject...) -> Results<T>
+-    public func filter(predicate: NSPredicate) -> Results<T>
+-    public func sorted(property: String, ascending: Bool = true) -> Results<T>
+-    public func sorted<S: SequenceType where S.Generator.Element == SortDescriptor>(sortDescriptors: S) -> Results<T>
+-    public func min<U: MinMaxType>(property: String) -> U?
+-    public func max<U: MinMaxType>(property: String) -> U?
+-    public func sum<U: AddableType>(property: String) -> U
+-    public func average<U: AddableType>(property: String) -> U?
+-    public func addNotificationBlock(block: (RealmCollectionChange<LinkingObjects> -> Void)) -> NotificationToken
+-    public func generate() -> RLMGenerator<T>
++    public override func value(forKey key: String) -> Any?
++    public override func value(forKeyPath keyPath: String) -> Any?
++    public override func setValue(_ value: Any?, forKey key: String)
++    public func filter(using predicateFormat: String, _ args: Any...) -> Results<T>
```

Comment:
This is one of the places I'm torn on. It's clear that there are a lot of users who don't realize that the argument to filter is an NSPredicate format string, so something that makes that obvious could be very beneficial. The problem is that I don't know how to communicate that without making it incredibly unwieldy, and I'm not sure what percentage of those people also don't know what a predicate format string is at all. All the people who use string interpolation or `stringWithFormat:` to build a string to pass to `predicateWithFormat:` obviously won't be helped by knowing that we pass the string to that.